substrate/client/consensus/aura/src/import_queue.rs

```diff
@@ -347,11 +350,81 @@ where
 		check_for_equivocation,
 		telemetry,
 		compatibility_mode,
-	});
+	})
+	.map_err(|e| sp_consensus::Error::Other(e.into()))?;
+
+	let authorities_tracker = verifier.authorities_tracker.clone();
+
+	let block_import = AuraBlockImport { block_import, authorities_tracker };
 
 	Ok(BasicQueue::new(verifier, Box::new(block_import), justification_import, spawner, registry))
 }
 
+/// AURA block import.
+pub struct AuraBlockImport<Client, P: Pair, Block: BlockT, BI: BlockImport<Block>> {
+	block_import: BI,
+	authorities_tracker: Arc<AuthoritiesTracker<P, Block, Client>>,
+}
+
+impl<Client, P: Pair, Block: BlockT, BI: BlockImport<Block>> AuraBlockImport<Client, P, Block, BI> {
+	/// Create a new AURA block import.
+	pub fn new(
+		block_import: BI,
+		authorities_tracker: Arc<AuthoritiesTracker<P, Block, Client>>,
+	) -> Self {
+		Self { block_import, authorities_tracker }
+	}
+}
+
+impl<Client, P: Pair, Block: BlockT, BI: BlockImport<Block> + Clone> Clone
+	for AuraBlockImport<Client, P, Block, BI>
+{
+	fn clone(&self) -> Self {
+		Self {
+			block_import: self.block_import.clone(),
+			authorities_tracker: self.authorities_tracker.clone(),
+		}
+	}
+}
+
+#[async_trait::async_trait]
+impl<Client: Sync + Send, P: Pair, Block: BlockT, BI: BlockImport<Block> + Send + Sync>
+	BlockImport<Block> for AuraBlockImport<Client, P, Block, BI>
+where
+	Client: HeaderBackend<Block>
+		+ HeaderMetadata<Block, Error = sp_blockchain::Error>
+		+ ProvideRuntimeApi<Block>,
+	P::Public: Codec + Debug,
+	P::Signature: Codec,
+	Client::Api: AuraApi<Block, AuthorityId<P>>,
+{
+	type Error = BI::Error;
+
+	async fn check_block(
+		&self,
+		block: BlockCheckParams<Block>,
+	) -> Result<ImportResult, Self::Error> {
+		self.block_import.check_block(block).await
+	}
+
+	/// Import a block.
+	async fn import_block(
+		&self,
+		block: BlockImportParams<Block>,
+	) -> Result<ImportResult, Self::Error> {
+		let post_header = block.post_header();
+		let import_from_runtime = matches!(block.state_action, StateAction::ApplyChanges(_));
```

Comment:
```suggestion
		let import_from_runtime = block.with_state();
```

Reply:
I think I understand what you mean, and I think you're right. We can just call `import` here instead of `import_from_runtime`. The origin of the problem we were having in production in https://github.com/paritytech/polkadot-sdk/issues/10007 is that the importing of authorities used to happen during verification rather than during block import, but now that this is no longer the case, we're good. @skunert Can you please sanity check this? Am I right? Am I missing something?